website/docs/rbs-support.md

```diff
@@ -34,7 +35,179 @@ def foo(x)
 end
 ```
 
-### Quick reference
+## Caveats
+
+> Support for this feature is experimental, and we actively discourage depending
+> on it for anything other than to offer feedback to the Sorbet developers.
+
+There are numerous shortcomings of the comment-based syntax versus `sig` syntax.
+By contrast, the near-sole upside is that the syntax is terser.
+
+### The comment-based syntax is second class
+
+The comment-based syntax uses [RBS syntax](https://github.com/ruby/rbs). RBS is
+an alternative annotation syntax for Ruby. The headline features of RBS:
+
+- It's the type annotation syntax blessed by the Ruby core team, and Matz in
+  particular (the lead of the Ruby project).
+- RBS supports certain enticing features, like method overloading and duck
+  types.
+- The syntax is terse.
+- The Ruby distribution bundles a gem for parsing RBS annotations.
+
+However, there are a number of problems which mean that RBS syntax has a
+second-class position in Sorbet:
+
+#### RBS syntax does not match the semantics of Sorbet
+
+Sorbet's type annotation syntax evolved differently from the evolution of RBS
+syntax. Unfortunately, this difference is not syntax-deep: it affects the
+semantics of the types too.
+
+Sure, there are similarities: both syntaxes offer a way to express union types,
+for example. But there are also semantic differences, and RBS syntax reflects
+these differences. Things which are possible to express in RBS syntax have no
+analogue in Sorbet and vice versa. Some examples:
+
+- RBS supports duck typing, but
+  [Sorbet does not](faq.md#can-i-use-sorbet-for-duck-typed-code), by design
+- Sorbet allows singleton classes to be generic (e.g.
+  [type_template](generics.md#type_member--type_template)). RBS has no such
+  concept.
+- Sorbet treats Ruby's `::Class` as a generic class, allowing sophisticated
+  class-level metaprogramming. RBS does not have syntax to represent this.
```

Comment:
I'm not sure I'm following this one, you can write `Class[Foo]`, we translate it as `T::Class[Foo]`: https://sorbet.run/?arg=--enable-experimental-rbs-signatures#%23%20typed%3A%20true%0A%0A%23%3A%20-%3E%20Class%5BInteger%5D%0Adef%20foo%3B%20T.unsafe%28nil%29%3B%20end%0A%0AT.reveal_type%28foo%29